src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,72 @@
 'use strict';
 
+const NAME = '';
+const HEALTH = 100;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = NAME, health = HEALTH) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
+const HIDDEN = false;
+
 class Herbivore extends Animal {
-  // write your code here
+  hidden;
+
+  constructor(name, health, hidden = HIDDEN) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  takeDamage(damage) {
+    if (!this.hidden) {
+      super.takeDamage(damage);
+    }
```

Comment:
The 'takeDamage' method in the 'Herbivore' class is correctly checking if the animal is hidden before applying damage. However, it would be more appropriate to move this logic to the 'bite' method of the 'Carnivore' class to encapsulate the biting behavior within the 'Carnivore' class.